src/frontend/components/UI/ActionIcons/index.tsx

```diff
@@ -14,55 +14,38 @@ import {
 } from '@fortawesome/free-regular-svg-icons'
 
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import React, { useCallback, useContext } from 'react'
+import React, { useContext } from 'react'
 import { useTranslation } from 'react-i18next'
 import ContextProvider from 'frontend/state/ContextProvider'
 import FormControl from '../FormControl'
 import './index.css'
 import { Runner } from 'common/types'
 import { Visibility, VisibilityOff } from '@mui/icons-material'
 import classNames from 'classnames'
+import LibraryContext from 'frontend/screens/Library/LibraryContext'
 
 interface Props {
-  sortDescending: boolean
-  sortInstalled: boolean
-  toggleSortDescending: () => void
-  toggleSortinstalled: () => void
   library: Runner | 'all'
 }
 
-export default React.memo(function ActionIcons({
-  library,
-  sortDescending,
-  toggleSortDescending,
-  sortInstalled,
-  toggleSortinstalled
-}: Props) {
+export default React.memo(function ActionIcons({ library }: Props) {
   const { t } = useTranslation()
+  const { refreshLibrary, refreshing } = useContext(ContextProvider)
+
   const {
-    refreshLibrary,
     handleLayout,
     layout,
     showHidden,
     setShowHidden,
     showFavourites,
-    refreshing,
     setShowFavourites,
     showNonAvailable,
-    setShowNonAvailable
-  } = useContext(ContextProvider)
-
-  const toggleNonAvailable = useCallback(() => {
-    setShowNonAvailable(!showNonAvailable)
-  }, [showNonAvailable])
-
-  const toggleShowHidden = useCallback(() => {
-    setShowHidden(!showHidden)
-  }, [showHidden])
-
-  const toggleShowFavourites = useCallback(() => {
-    setShowFavourites(!showFavourites)
-  }, [showFavourites])
+    setShowNonAvailable,
+    sortDescending,
+    setSortDescending,
+    sortInstalled,
+    setSortInstalled
+  } = useContext(LibraryContext)
```

Comment:
some state came from the global context, some from props, now it's all from the library context